lib/web/databases.go

```diff
@@ -141,16 +117,40 @@ func (h *Handler) handleDatabaseCreate(w http.ResponseWriter, r *http.Request, p
 
 // updateDatabaseRequest contains some updatable fields of a database resource.
 type updateDatabaseRequest struct {
-	CACert string `json:"ca_cert,omitempty"`
+	// CACert when non-nil means that this request is to
+	// only update the CA and validates it.
+	// When nil, skips validating CACert and updates all other
+	// fields.
```

Comment:
I wouldn't turn this into an OR (update CACert OR other fields)
If any of the other fields is present, then they should be updated

We can use the same method for the other fields: if nil, then ignore its value and update the other fields.